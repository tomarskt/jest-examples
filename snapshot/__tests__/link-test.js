import React from 'react'
import renderer from 'react-test-renderer'
import Link from '../Link'


jest.mock('frankel-uk/Icon', ()=>'Icon', {virtual: true}) //if you have this line, then you will have <Icon tpe='facebook' /> instead of <div><img src='../img.png'/></div> . i.e. Icon is mocked
// no need to call jest.genMockFromModule('fs') either ; 
// no need to create a 'module_name.js' under __mocks__ folder
describe('Link', () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <Link href="http://facebook.com/" icon="facebook">
        Facebook
      </Link>
    )

    expect(component.toJSON()).toMatchSnapshot()
   })

})