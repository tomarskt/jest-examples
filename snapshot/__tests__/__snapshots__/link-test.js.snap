// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link should render correctly 1`] = `
<div>
  <Icon
    type="facebook"
  />
  Facebook
</div>
`;
